fk/secretreceive.go

```diff
@@ -202,13 +230,77 @@ func countDigits(i int) (count int) {
 	return len(iString)
 }
 
+func getFileNameToWrite(filename string) (string, error) {
+	userDir, err := homedir.Dir()
+	if err != nil {
+		return "", err
+	}
+	downloadsDir := userDir + "/Downloads"
+	fileInfo, err := os.Stat(downloadsDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return "", fmt.Errorf(downloadsDir + " directory doesn't exist")
+		} else {
+			return "", err
+		}
+	}
+	if !fileInfo.IsDir() {
+		return "", fmt.Errorf(fileInfo.Name() + " file exists and is not a directory")
+	}
+	filenameToWrite := getNewUniqueFilename(
+		filepath.FromSlash(downloadsDir+"/"+filename),
+		doesntExist,
+	)
+
+	return filenameToWrite, nil
+}
+
+func getNewUniqueFilename(fp string, checker func(string) bool) string {
+	i := 0
+	foundFilename := false
+	filename := filepath.Base(fp)
+	numberedFilename := filename
+	for ok := true; ok; ok = (foundFilename == false) {
+		if i == 0 {
+			// try using the original filename it was sent as first with no count suffix
+			numberedFilename = filename
+		} else if i > 0 {
+			// the original filename already exists, try `filename(i).ext` next...
+			numberedFilename = fmt.Sprintf(
+				"%s(%d)%s",
+				FilenameWithoutExtension(filename), i, path.Ext(filename),
+			)
+		}
+		if checker(filepath.FromSlash(filepath.Dir(fp) + "/" + numberedFilename)) {
+			foundFilename = true
+		} else {
+			i++
+		}
+	}
+	return numberedFilename
+}
+
+func doesntExist(path string) bool {
```

Comment:
Yep I see what you're doing: I think it would be clearer if this was positive though: `fileExists` (and I think we may have that code elsewhere, so ripe for a utility)

Reply:
Ok!

Reply:
I ended up setting on `IsSafeToWrite` which is more explicit (it's not actually `!fileExists` because what if it exists AND it cannot be written?)